src/org/thoughtcrime/securesms/profiles/AvatarHelper.java

```diff
@@ -1,95 +1,58 @@
         DcContext dcContext = DcHelper.getContext(context);
-        String avatarPath = dcContext.getChat(chatId).getProfileImage();
-        if (avatarPath != null && !avatarPath.isEmpty()) {
-            File oldImage = new File(avatarPath);
-            if (oldImage.exists()) {
+
+        if (bitmap == null) {
+            dcContext.setChatProfileImage(chatId, null);
+        } else {
+            try {
+                File avatar = File.createTempFile("groupavatar", "jpg", context.getCacheDir());
+                FileOutputStream out = new FileOutputStream(avatar);
+                bitmap.compress(Bitmap.CompressFormat.JPEG, 85, out);
+                out.close();
+                dcContext.setChatProfileImage(chatId, avatar.getPath()); // The avatar is copied to the blobs directory here...
                 //noinspection ResultOfMethodCallIgnored
-                oldImage.delete();
+                avatar.delete(); // ..., now we can delete it.
+            } catch (IOException e) {
+                e.printStackTrace();
             }
         }
-        avatarPath = getFilePathForGroupAvatar(context, chatId, System.currentTimeMillis());
-        FileOutputStream outStream;
-        try {
-            outStream = new FileOutputStream(avatarPath);
-            bitmap.compress(Bitmap.CompressFormat.JPEG, 85, outStream);
-            dcContext.setChatProfileImage(chatId, avatarPath);
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
-        }
     }
 
-    public static InputStream getInputStreamFor(@NonNull Context context, @NonNull String address)
-            throws IOException {
-        return new FileInputStream(getSelfAvatarFile(context, address));
+    public static File getSelfAvatarFile(@NonNull Context context) {
+        String dirString = DcHelper.getContext(context).getConfig(DcHelper.CONFIG_SELF_AVATAR);
+        return new File(dirString);
     }
 
-    public static File getSelfAvatarFile(@NonNull Context context, @NonNull Address address) {
-        String name = new File(address.serialize()).getName();
-        return getSelfAvatarFile(context, name);
-    }
-
-    public static File getSelfAvatarFile(@NonNull Context context, @NonNull String address) {
-        File avatarDirectory = new File(context.getFilesDir(), AVATAR_DIRECTORY);
-        avatarDirectory.mkdirs();
-        return new File(avatarDirectory, address+".jpg");
-    }
-
-    public static void setSelfAvatar(@NonNull Context context, @NonNull String address, @Nullable byte[] data) throws IOException {
-        File avatar = getSelfAvatarFile(context, address);
+    public static void setSelfAvatar(@NonNull Context context, @Nullable byte[] data) throws IOException {
         if (data == null) {
-            avatar.delete();
             DcHelper.set(context, DcHelper.CONFIG_SELF_AVATAR, null);
         } else {
+            File avatar = File.createTempFile("selfavatar", "jpg", context.getCacheDir());
```

Comment:
same here: the extension needs to be `.jpg` - otherwise, at least the control showing it has problems.